src/terminal/input/terminalInput.cpp

```diff
@@ -466,6 +468,45 @@ bool TerminalInput::HandleKey(const IInputEvent* const pInEvent) const
     return fKeyHandled;
 }
 
+bool TerminalInput::HandleChar(const wchar_t ch)
+{
+    if (ch == UNICODE_BACKSPACE || ch == UNICODE_DEL)
+    {
+        return false;
+    }
+    else if (Utf16Parser::IsLeadingSurrogate(ch))
+    {
+        if (_leadingSurrogate.has_value())
+        {
+            // we already were storing a leading surrogate but we got another one. Go ahead and send the
+            // saved surrogate piece and save the new one
+            wchar_t buffer[32];
+            swprintf_s(buffer, L"%I32u", _leadingSurrogate.value());
+            _SendInputSequence(buffer);

```

Comment:
Sorry, this is a bit nuanced. This probably compiled correctly because there's a direct creation of a `std::wstring_view` from a `wchar_t[]`, but it's going to end up setting the length to the buffer size, most likely, causing an overrun into uninitialized space. In this specific situation, we probably need to count the actual written length into the buffer (might be a return code from `swprintf_s`) and manually allocate the `std::wstring_view` with that information to prevent an overrun now that we're not counting.

That's sort of why I said there should be two definitions: one that helps count and one that already knows the count. But if we can detect/pass the count in here with an explicit `std::wstring_view` construction (presuming we know it or wcslen it here), then we only need the one definition of `_SendInputSequence` with the `std::wstring_view`.

Reply:
UPDATE: I checked and the return code of `swprintf_s` should be the length written if it was valid.

So something like
```
const auto written = swprintf_s(buffer, L"%I32u", _leadingSurrogate.value());
_SendInputSequence({buffer, written}); 
```

is probably sufficient for giving the correct length data and preventing an overrun, avoiding counting entirely, and not needing two definitions of `_SendInputSequence`

Reply:
I thought from reading the C++ reference that when it creates the `std::wstring_view` from a buffer it ends up counting to the null terminating character, which `swprintf` should put into the buffer. Namely [this](https://en.cppreference.com/w/cpp/string/char_traits/length) doc on how the length is determined for constructor 4 on [this](https://en.cppreference.com/w/cpp/string/basic_string_view/basic_string_view) page.

Reply:
Oh snap. You're right. So it was correct the whole time. Gap in my knowledge, sorry about that.

It does look like it does still do the count operation though to seek forward to where the Z is to count it which you could save with my suggestion. But I'm not going to press it unless you're particularly excited about squeezing every ounce out of this by not counting it. You can resolve this as is if you like.

Reply:
I'll leave it as-is for now. This particular segment of code should only get hit when we get malformed UTF-16 sequences so the perf impact should be minimal. If in the future we determine its an issue we can fix it then.